src/components/ApiReference/NavItem.js

```diff
@@ -0,0 +1,105 @@
+import React from "react";
+import styled, { css } from "styled-components";
+import PropTypes from "prop-types";
+
+import {
+  CSS_TRANSITION_SPEED,
+  FONT_WEIGHT,
+  PALETTE,
+  MEDIA_QUERIES,
+} from "constants/styles";
+
+import { isInViewport } from "helpers/dom";
+import { useMatchMedia } from "helpers/useMatchMedia";
+
+import { Context as ScrollRouterContext } from "components/ApiReference/ScrollRouter";
+
+const activeStyles = `
+  color: ${PALETTE.purpleBlue};
+  font-weight: ${FONT_WEIGHT.bold};
+`;
+const NavItemEl = styled.div`
+  text-align: left;
+  white-space: nowrap;
+  font-size: ${(props) => (props.depth > 0 ? "0.875rem" : "1rem")};
+  color: ${(props) => (props.depth === 0 ? PALETTE.black80 : PALETTE.black60)};
+  padding: 0.25rem 0;
+  padding-left: ${(props) => (props.depth > 1 ? `${props.depth - 1}rem` : 0)};
+  transition: opacity ${CSS_TRANSITION_SPEED.default} ease-out;
+  font-weight: ${FONT_WEIGHT.normal};
+
+  ${(props) =>
+    props.isActive
+      ? css`
+          ${activeStyles}
+        `
+      : ""}
+`;
+
+const NAV_BAR_HEIGHT = 89;
+const FIXED_NAV_DISTANCE = 140 + NAV_BAR_HEIGHT;
+export const NavItem = ({ isActive, forwardedRef, children, depth }) => {
```

Comment:
I peel this out of `templates/ApiReference.js` both to unify styles on the single/full API reference pages, and because the template was fairly hard to navigate. Template had a lot of things all living in one place.